chore(flags): Add dual-write support for dedicated Redis cache

haacked opened the pull request:
Implements dual-write functionality for feature flags cache to support migration from shared to dedicated Redis for flags.

We always write to both caches if the dedicated cache is configured. This is because the Django app will always read from the shared cache, while the Rust service will read from the dedicated cache if available and enabled.

1. `FLAGS_REDIS_URL` is not set (no dedicated flags cache)  - writes to shared cache only
2. `FLAGS_REDIS_URL` is set (dedicated flags cache) - writes to both shared and dedicated cache.
   - Rust `/flags` service code reads from dedicated cache
   - Django code (`/my_flags` and `/evaluation_reasons` reads from shared cache)

## Problem

We need flag changes to invalidate the flags cache in the Flags Dedicated Redis.

## Changes

Adds a new setting to configure the dedicated flags Redis cache.

* `FLAGS_REDIS_URL` - dedicated flags Redis cache with write access (Django only ever writes to this cache).

## How did you test this code?

Unit Tests
Manual testing

### Test Scenarios:

Each scenario is tested by modifying the `.env` file. Each time I call `redis-cli FLUSHALL` to clear the cache. I then run `bin/start --minimal` and edit a flag in the UI.

1. Current behavior

```yaml
# FLAGS_REDIS_URL=redis://localhost:6379/1
```

```yaml
# Invalid values act as if this wasn't set.
# FLAGS_REDIS_URL=xxxx or redis://localhost:0000/1 or xxxx://localhost:6379/1
```

Expected:
1. `redis-cli -n 0 GET posthog:1:team_feature_flags_1` returns flags
2. `redis-cli -n 1 GET posthog:1:team_feature_flags_1` returns nothing.


2. Dual Write

```yaml
FLAGS_REDIS_URL=redis://localhost:6379/1
```

Expected:
1. `redis-cli -n 0 GET posthog:1:team_feature_flags_1` returns flags.
2. `redis-cli -n 1 GET posthog:1:team_feature_flags_1` returns flags.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->
<!-- Removing this section does not mean the changelog bot won't pick it up, because *some people* like to not use the template, so we can't rely on it existing. -->

dmarticus reviewed (approved):
On top of the code-level reviews, here's my take on your open question: Yes, always dual-write even when FLAGS_REDIS_ENABLED=true. The whole point of this cache design is that we're isolating the high-scale external `/flags` endpoint, but not all flag evaluation operations. The internal app endpoints (`/my_flags`, `/evaluation_reasons`) have much lower traffic and can safely share Redis with other Django operations. Dual-writing is the correct permanent architecture, IMO.

haacked commented on posthog/settings/data_stores.py:
```diff
@@ -419,5 +425,23 @@ def _parse_kafka_hosts(hosts_string: str) -> list[str]:
+#    - Writes to both shared (source of truth) and dedicated (warming) caches
+# 3. Dedicated Redis configured, FLAGS_REDIS_ENABLED=true: Uses dedicated cache only
+if FLAGS_REDIS_URL:
+    CACHES["flags_dedicated"] = {
```
Looking through the code:

1. Settings load (data_stores.py:435-444) - Just creates a Python dictionary with config. No Redis code runs.
2. Django cache system - Doesn't instantiate backends until first use
3. `django-redis` RedisCache - Doesn't call `redis.from_url()` until you actually call `cache.get()` or `cache.set()`
4. `redis-py` - Only then does it parse the URL and validate the scheme

haacked committed:
Add dual-write support for dedicated Redis cache

Implements dual-write functionality for feature flags cache to support migration from shared to dedicated Redis for flags.

We always write to both caches if the dedicated cache is configured. This is because the Django app will always read from the shared cache, while the Rust service will read from the dedicated cache if available and enabled.

1. `FLAGS_REDIS_URL` is not set (no dedicated flags cache)  - writes to shared cache only
2. `FLAGS_REDIS_URL` is set (dedicated flags cache) - writes to both shared and dedicated cache.

Simplify Django cache configuration to remove unused settings

Django never reads from the dedicated flags cache, only writes to it.

haacked commented:
Actually, it occurs to me we can simplify the Django code even more because the Django code never needs to read from the dedicated cache. It only ever writes to it. So I got rid of the `FLAGS_REDIS_ENABLED` on the Django side.
